client/ws.ts

```diff
@@ -59,50 +79,70 @@ export class Remote {
         }
         const parsedData = JSON.parse(payloadData);
 
-        //Processes for the method 'emitBatch':
-        if (Array.isArray(parsedData) && !isOnetime && parsedData.length > 0) {
-          const invalid = parsedData.map(validateResponse).find((res) =>
+        // Process for the method 'emitBatch':
+        if (Array.isArray(parsedData) && parsedData.length > 0) {
+          const rpcResponses = parsedData.map(validateResponse);
+          const invalid = rpcResponses.find((res) =>
             !isObject(res.result) || res.result.event !== "emitted"
           );
           if (invalid) {
             throw new BadServerDataError(
-              invalid.id ? invalid.id : null,
+              invalid.id || null,
               "The server returned an invalid batch response.",
               -32004,
             );
           } else {
-            continue;
+            for (const res of rpcResponses) {
+              if ((res.result as JsonObject).id === rpcRequest.id) {
+                yield res.result;
+              }
+            }
```

Comment:
I'm not 100% sure about this. But I am right that before this change the subscription event were just never outputted anywhere? They were just skipped? What was the point of a subscription then?